server/room-battle.js

```diff
@@ -783,7 +783,15 @@ class RoomBattle extends RoomGames.RoomGame {
 			parentGame.onBattleWin(this.room, winnerid);
 		}
 		// If the room's replay was hidden, disable users from joining after the game is over
-		if (this.room.hideReplay) this.room.modjoin = '%';
+		if (this.room.hideReplay) {
+			this.room.modjoin = '%';
+			this.room.isPrivate = 'hidden';
+			// We also save the replay to make sure the hidden setting gets reflected
```

Comment:
I would only do this if the replay already exists.

Reply:
How do I determine if the replay already exists?

Reply:
Add a new property on the battle or room (probably battle) to track whether or not the replay has been saved.